tests/ctl/cli/test_cli.py

```diff
@@ -169,678 +171,735 @@ def test_audit(test_config_path: str, test_cli_runner: CliRunner) -> None:
+        )
+        with open(tmp_keyfile, "w", encoding="utf-8") as keyfile:
+            dump(config_data_decoded, keyfile)
+        dataset_name = "fidesopstest"
+        result = test_cli_runner.invoke(
+            cli,
+            [
+                "-f",
+                test_config_path,
+                "generate",
+                "dataset",
+                "gcp",
+                "bigquery",
+                dataset_name,
+                f"{tmp_output_file}",
+                "--keyfile-path",
+                f"{tmp_keyfile}",
+            ],
+        )
+        print(result.output)
+        assert result.exit_code == 0
 
-@pytest.mark.external
-def test_scan_system_okta_input_credentials_id(
-    test_config_path: str,
-    test_cli_runner: CliRunner,
-) -> None:
-    os.environ["FIDES__CREDENTIALS__OKTA_1__TOKEN"] = os.environ["OKTA_CLIENT_TOKEN"]
-    result = test_cli_runner.invoke(
-        cli,
-        [
-            "-f",
-            test_config_path,
-            "scan",
-            "system",
-            "okta",
-            "--credentials-id",
-            "okta_1",
-            "--coverage-threshold",
-            "0",
-        ],
-    )
-    print(result.output)
-    assert result.exit_code == 0
 
+class TestUser:
+    """
+    Test the "user" command group.
 
-@pytest.mark.external
-def test_generate_dataset_bigquery_credentials_id(
-    test_config_path: str,
-    test_cli_runner: CliRunner,
-    tmpdir: LocalPath,
-) -> None:
+    Most tests rely on previous tests.
+    """
 
-    tmp_output_file = tmpdir.join("dataset.yml")
-    config_data = os.getenv("BIGQUERY_CONFIG", "e30=")
-    config_data_decoded = loads(b64decode(config_data.encode("utf-8")).decode("utf-8"))
-    os.environ["FIDES__CREDENTIALS__BIGQUERY_1__PROJECT_ID"] = config_data_decoded[
-        "project_id"
-    ]
-    os.environ["FIDES__CREDENTIALS__BIGQUERY_1__PRIVATE_KEY_ID"] = config_data_decoded[
-        "private_key_id"
-    ]
-    os.environ["FIDES__CREDENTIALS__BIGQUERY_1__PRIVATE_KEY"] = config_data_decoded[
-        "private_key"
-    ]
-    os.environ["FIDES__CREDENTIALS__BIGQUERY_1__CLIENT_EMAIL"] = config_data_decoded[
-        "client_email"
-    ]
-    os.environ["FIDES__CREDENTIALS__BIGQUERY_1__CLIENT_ID"] = config_data_decoded[
-        "client_id"
-    ]
-    os.environ[
-        "FIDES__CREDENTIALS__BIGQUERY_1__CLIENT_X509_CERT_URL"
-    ] = config_data_decoded["client_x509_cert_url"]
-    dataset_name = "fidesopstest"
-    result = test_cli_runner.invoke(
-        cli,
-        [
-            "-f",
-            test_config_path,
-            "generate",
-            "dataset",
-            "gcp",
-            "bigquery",
-            dataset_name,
-            f"{tmp_output_file}",
-            "--credentials-id",
-            "bigquery_1",
-        ],
-    )
-    print(result.output)
-    assert result.exit_code == 0
+    @pytest.mark.unit
+    def test_user_login_provide_credentials(
```

Comment:
I think we should add a way to patch `CREDENTIALS_FILE_PATH` at test time and use a fixture to patch the path with `tmp_path` so we don't write test data to user's actual home directory.

Reply:
that's a `to do` up above :) working on it today